Lib/typing.py

```diff
@@ -1285,6 +1494,38 @@ def copy_with(self, params):
         return super().copy_with(params)
 
 
+@_SpecialForm
+def Unpack(self, parameters):
+    """Implementation of the type unpack operator for older versions of Python.
+
+    The type unpack operator takes the child types from some container type
+    such as `Tuple[int, str]` and "pulls them out". For example,
+    `Dict[Unpack[Tuple[int, str]]` is equivalent to `Dict[int, str]`.
+    In newer versions of Python, this is implemented using the `*` operator.
```

Comment:
Ah, good point.

`Array` might not be the best choice either, because some readers might not be familiar with the numerical computing context, so I've revised this, just pointing to a "generic class `Foo`".